src/query/range_query_ip_fastfield.rs

```diff
@@ -0,0 +1,197 @@
+        match field_type.fastfield_cardinality().unwrap() {
+            Cardinality::SingleValue => {
+                let ip_addr_fast_field = reader.fast_fields().ip_addr(self.field)?;
+                let value_range =
+                    bound_to_value_range(&self.left_bound, &self.right_bound, &ip_addr_fast_field);
+                let docset = IpRangeDocSet::new(value_range, ip_addr_fast_field);
+                Ok(Box::new(ConstScorer::new(docset, boost)))
+            }
+            Cardinality::MultiValues => unimplemented!(),
+        }
+    }
+
+    fn explain(&self, reader: &SegmentReader, doc: DocId) -> crate::Result<Explanation> {
+        let mut scorer = self.scorer(reader, 1.0)?;
+        if scorer.seek(doc) != doc {
+            return Err(TantivyError::InvalidArgument(format!(
+                "Document #({}) does not match",
+                doc
+            )));
+        }
+        let explanation = Explanation::new("Const", scorer.score());
+
+        Ok(explanation)
+    }
+}
+
+fn bound_to_value_range(
+    left_bound: &Bound<Ipv6Addr>,
+    right_bound: &Bound<Ipv6Addr>,
+    column: &Arc<dyn Column<Ipv6Addr>>,
+) -> RangeInclusive<Ipv6Addr> {
+    let start_value = match left_bound {
+        Bound::Included(ip_addr) => *ip_addr,
+        Bound::Excluded(ip_addr) => Ipv6Addr::from(ip_addr.to_u128() + 1),
+        Bound::Unbounded => column.min_value(),
+    };
+
+    let end_value = match right_bound {
+        Bound::Included(ip_addr) => *ip_addr,
+        Bound::Excluded(ip_addr) => Ipv6Addr::from(ip_addr.to_u128() - 1),
+        Bound::Unbounded => column.max_value(),
+    };
+    start_value..=end_value
+}
+
+/// Helper to have a cursor over a vec
+struct VecCursor {
+    data: Vec<u64>,
+    pos_in_data: usize,
+}
+impl VecCursor {
+    fn new() -> Self {
+        Self {
+            data: Vec::with_capacity(32),
+            pos_in_data: 0,
+        }
+    }
+    fn next(&mut self) -> Option<u32> {
+        self.pos_in_data += 1;
+        self.current()
+    }
+    fn current(&self) -> Option<u32> {
+        self.data.get(self.pos_in_data).map(|el| *el as u32)
+    }
+
+    fn set_data(&mut self, data: Vec<u64>) {
+        self.data = data;
+        self.pos_in_data = 0;
+    }
+    fn is_empty(&self) -> bool {
+        self.pos_in_data >= self.data.len()
+    }
+}
+
+struct IpRangeDocSet {
+    value_range: RangeInclusive<Ipv6Addr>,
+    ip_addr_fast_field: Arc<dyn Column<Ipv6Addr>>,
+    fetched_until_doc: u32,
+    loaded_docs: VecCursor,
+}
+impl IpRangeDocSet {
+    fn new(
+        value_range: RangeInclusive<Ipv6Addr>,
+        ip_addr_fast_field: Arc<dyn Column<Ipv6Addr>>,
+    ) -> Self {
+        let mut ip_range_docset = Self {
+            value_range,
+            ip_addr_fast_field,
+            loaded_docs: VecCursor::new(),
+            fetched_until_doc: 0,
+        };
+        ip_range_docset.fetch_block();
+        ip_range_docset
+    }
+
+    /// Returns true if more data could be fetched
```

Comment:
```suggestion
    /// Returns true if more data could be fetched
```
the comment is wrong